test/functional/rpc_packages.py

```diff
@@ -369,6 +369,13 @@ def test_submit_child_with_parents(self, num_parents, partial_submit):
     def test_submitpackage(self):
         node = self.nodes[0]
 
+        self.log.info("Submitpackage only allows valid hex inputs")
+        valid_tx_list = self.wallet.create_self_transfer_chain(chain_length=2)
+        hex_list = [valid_tx_list[0]["hex"][:-1] + 'X', valid_tx_list[1]["hex"]]
+        txid_list = [valid_tx_list[0]["txid"], valid_tx_list[1]["txid"]]
+        assert_raises_rpc_error(-22, "TX decode failed:", node.submitpackage, hex_list)
+        assert txid_list[0] not in node.getrawmempool()
```

Comment:
```suggestion
        assert txid_list[0] not in node.getrawmempool()
        assert txid_list[1] not in node.getrawmempool()
```

Reply:
Thanks! 

Updated in d7fd766feb2f579bdba0e778bacdeb13103e8282